registry/api/v2/urls.go

```diff
@@ -72,6 +74,18 @@ func NewURLBuilderFromRequest(r *http.Request, relative bool) *URLBuilder {
 		host = strings.TrimSpace(hosts[0])
 	}
 
+	forwardedPort := r.Header.Get("X-Forwarded-Port")
```

Comment:
But, a port in the `Host` header _is_ explicit, in that it is guaranteed to be set by a single entity, whereas `X-Forward-Port` may have been set by a secondary party.